tests/test_variable.py

```diff
@@ -30,80 +28,55 @@
+    assert all([m * p == (m * v).val for p, v, m in zip(par_free, var, mul)])  # left scalar mul
+    assert all([m * p == (v * m).val for p, v, m in zip(par_free, var, mul)])  # right scalar mul
+    assert all([p / m == pytest.approx((v / m).val) for p, v, m in zip(par_free, var, mul)])  # right scalar div
+    assert all([-p == (-v).val for p, v in zip(par_free, var)])  # unary minus
+    assert all([-m**2 * p == (m * -v * m).val for p, v, m in zip(par_free, var, mul)])  # compound expression
 
 
 def test_keyword_variable():
     """
-    variable: Keyword Variable reference tests.\
+    variable: Keyword Variable reference tests.
     """
-    n = 10
-    m = nr.randn(n)  # parameter multipliers
-    par_fixed = nr.randn(n)  # fixed parameter values
-    par_free = nr.randn(n)  # free parameter values
-    Variable.kwarg_values = {k: v for k, v in zip(ascii_lowercase, par_free)}
-
-    # test __str__()
-    p = Variable(0, name="kw1")
-    assert str(p) == "Variable 0: name = kw1, "
-    assert str(-p) == "Variable 0: name = kw1,  * -1"
-    assert str(1.2 * p * 0.4) == "Variable 0: name = kw1,  * 0.48"
-    assert str(1.2 * p / 2.5) == "Variable 0: name = kw1,  * 0.48"
-
-    # mapping function must yield the correct parameter values
-    assert (
-        (par_free[k] * m[k]) == (m[k] * Variable(name=n))
-        for (k, n) in zip(range(n), ascii_lowercase)
-    )
-    assert (
-        (par_free[k] * m[k]) == (Variable(name=n) * m[k])
-        for (k, n) in zip(range(n), ascii_lowercase)
-    )
-    assert (
-        (-par_free[k] * m[k]) == (-Variable(name=n))
-        for (k, n) in zip(range(n), ascii_lowercase)
-    )
-    assert (
-        (-par_free[k] * m[k] ** 2) == (m[k] * -Variable(name=n) * m[k])
-        for (k, n) in zip(range(n), ascii_lowercase)
-    )
-
     # Check for a single kwarg_value
     Variable.kwarg_values = {"kw1": 1.0}
-    assert Variable(name="kw1").val == 1.0
+    assert Variable(0, name="kw1").val == 1.0
 
-    # fixed values remain constant
-    assert [(par_fixed[k] == par_fixed[k]) for k in range(n)]
+    names = string.ascii_lowercase
+    n = 10
+    mul = nr.randn(n)  # parameter multipliers
+    par_free = nr.randn(n)  # free parameter values
+    Variable.kwarg_values = {k: v for k, v in zip(names, par_free)}
+    var = [Variable(0, name=n) for n in names]
+
+    assert all([p == v.val for p, v in zip(par_free, var)])  # basic evaluation
+    assert all([m * p == (m * v).val for p, v, m in zip(par_free, var, mul)])  # left scalar mul
+    assert all([m * p == (v * m).val for p, v, m in zip(par_free, var, mul)])  # right scalar mul
+    assert all([p / m == pytest.approx((v / m).val) for p, v, m in zip(par_free, var, mul)])  # right scalar div
+    assert all([-p == (-v).val for p, v in zip(par_free, var)])  # unary minus
+    assert all([-m**2 * p == (m * -v * m).val for p, v, m in zip(par_free, var, mul)])  # compound expression
```

Comment:
Each of these asserts should ideally be separate tests, with the test data parametrized using pytest decorators.

Reply:
I parametrized them, but I don't really see the benefit of splitting each assert into a separate test. They test different aspects of the same feature.